db_lib.py

```diff
@@ -1,630 +1,234 @@
-    except AttributeError:
-        # not all containers have a fixed capacity, eg. shipping dewar.
-        # depends on what subcontainers are used...
-        print("no capacity")
 
-#    kwargs['item_list'] = []
+def updateContainer(cont_info):
+    cont = cont_info['uid']
+    q = {'uid': cont_info.pop('uid', '')}
+    container_ref.update(q, cont_info)
 
-    c = Container(**kwargs)
-    c.save()
+    return cont
 
-    return c.container_id
 
-
-def createSample(sample_name, sample_type, **kwargs):
+def createSample(sample_name, **kwargs):
     """
     sample_name:  string, name for the new sample, required
-    sample_type:  required, either:
-                  - string, name for existing type in Types collection, or
-                  - mongo type object for existing type in Types collection
     kwargs:       passed to constructor
     """
-    kwargs['sampleName'] = sample_name
-    kwargs['requestList'] = []
-    kwargs['resultList'] = []
-
     # initialize request count to zero
     if 'request_count' not in kwargs:
         kwargs['request_count'] = 0
 
-    if isinstance(sample_type, str) or isinstance(sample_type, str):
-        kwargs['sample_type'] = type_from_name(sample_type, as_mongo_obj=True)
-
-#    try:
-#        kwargs['sample_type'] = Types.objects(__raw__={'name': type_name})[0]
-#    except IndexError:
-#        raise ValueError('no sample type found matching "{0}"'.format(type_name))
-
-    s = Sample(**kwargs)
-    s.save()
-
-    return s.sample_id
+    uid = sample_ref.create(name=sample_name, **kwargs)
+    return uid
 
 
 def incrementSampleRequestCount(sample_id):
     """
     increment the 'request_count' attribute of the specified sample by 1
     """
-    
+
     # potential for race here?
-    s = Sample.objects(__raw__={'sample_id': sample_id})[0]
-    s.update(inc__request_count=1)
-    s.reload()
-    return s.request_count
+    sample_ref.update(query={'uid': sample_id}, update={'$inc': {'request_count': 1}})
+    return getSampleRequestCount(sample_id)
 
 
 def getSampleRequestCount(sample_id):
     """
     get the 'request_count' attribute of the specified sample
     """
-    
-    s = Sample.objects(__raw__={'sample_id': sample_id})
-    return _try0_dict_key(s, 'sample', 'sample_id', sample_id, None, 
-                           dict_key='request_count')
+    s = list(getSampleByID(sample_id))
+    return s[0]['request_count']
 
 
-def getRequestsBySampleID(sample_id):
+def getRequestsBySampleID(sample_id, active_only=True):
     """
     return a list of request dictionaries for the given sample_id
     """
-    s = getSampleByID(sample_id, as_mongo_obj=True)
-    return [r.to_mongo() for r in s.requestList]
+    params = {'sample': sample_id}
+    if active_only:
+        params['active'] = True
+    reqs = list(requests_ref.find(**params)
+    return reqs
 
 
-def getSampleByID(sample_id, as_mongo_obj=False):
+def getSampleByID(sample_id):
```

Comment:
just a note that the behavior looks like it's changed slightly...
- old version returned a single item or "None"
- new version returns a list, empty list if no sample_id matches?
looks like it's only used internal to db_lib though, so I assume it's self consistent.

although,
getSampleNamebyID looks like it expects a dict-ish return value?